src/codegen/code_context.cpp

```diff
@@ -314,14 +328,34 @@ bool CodeContext::Compile() {
   }
 
   // Log the module
+  LOG_TRACE("%s\n", GetIR().c_str());
   if (settings::SettingsManager::GetBool(settings::SettingId::dump_ir)) {
     LOG_DEBUG("%s\n", GetIR().c_str());
   }
+}
 
-  // All is well
-  return true;
+size_t CodeContext::GetTypeSize(llvm::Type *type) const {
```

Comment:
There is a bunch of these const getter functions that would actually make sense in `CodeGen`. However, I need some of them during bytecode generation, where the `CodeGen` object doesn't exist anymore (and the IR also shouldn't be changed anymore). 
Should I simply create another `CodeGen` instance in the bytecode translator?

Reply:
@pmenon I just saw that I can't create a CodeGen instance in the bytecode translator, because I only have a const reference to the CodeContext. I want it to remain const to make sure that the bytecode translator doesn't change the IR. 

What we can do:
1. Move the getter functions to CodeContext (where they can be called with a const reference)
2. Create another wrapper class e.g. CodeHelper that contains these functions. But since I stripped them down there are only 3 of them left: GetSize, GetElementOffset and GetElementOffsetInBits.

Reply:
I guess leave them in `CodeContext` and have `CodeGen` just wrap them, as appropriate.